File: Python - Technical Placement Training/rat_food.py
def food(r,unit,arr):
    required=r*unit
    sum=0
    for i in range(len(arr)):
        sum=sum+arr[i]
        if sum>required:
            print("Output: ",i+1)
            exit(1)
    if arr[0]=="":
        print("-1")
    else:
        print("0")

r=int(input("r: "))
unit=int(input("unit: "))
n=int(input("n: "))
arr=[]
print("arr: ",end='')
for i in range(0,n):
    num=int(input())
    arr.append(num)
food(r,unit,arr)

File: C - Technical Placement Training/rat_food.c
#include<stdio.h>
#include<stdlib.h>
void calculate(int r,int unit,int arr[],int n);

void main()
{
    int r,unit,n;
    printf("r: ");
    scanf("%d",&r);
    printf("unit: ");
    scanf("%d",&unit);
    printf("n: ");
    scanf("%d",&n);
    int arr[n];
    int i;
    printf("arr: ");
    for(i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }
    calculate(r,unit,arr,n);
}

void calculate(int r,int unit,int arr[],int n)
{
    int required_food = r*unit;
    int sum=0,i,done;
    for(i=0;i<n;i++)
    {
        sum=sum+arr[i];
        if(sum>=required_food)
        {
            done=i+1;
            printf("%d",done);
            exit(0);
        }
    }
    if(arr[0]==' ')
    {
        printf("-1");
    }
    else if(sum!=required_food)
    {
        printf("0");
    }
}